receiver/hostmetricsreceiver/hostmetrics_linux.go

```diff
@@ -0,0 +1,71 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+
+package hostmetricsreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/hostmetricsreceiver"
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+var gopsutilEnvVars = map[string]string{
+	"HOST_PROC": "/proc",
+	"HOST_SYS":  "/sys",
+	"HOST_ETC":  "/etc",
+	"HOST_VAR":  "/var",
+	"HOST_RUN":  "/run",
+	"HOST_DEV":  "/dev",
+}
+
+// This exists to validate that different instances of the hostmetricsreceiver do not
+// have inconsistent root_path configurations. The root_path is passed down to gopsutil
+// through env vars, so it must be consistent across the process.
+var globalRootPath string
```

Comment:
Is there a lock or atomic access on this value?

Just in case there several instances of it started at once?

Reply:
I considered a lock, but decided it was unnecessary complexity. I checked the access and it is not accessed concurrently. `globalRootPath` is accessed solely from `Validate()` which is called sequentially for all configs from [this line of code](https://github.com/open-telemetry/opentelemetry-collector/blob/d6ed8246b64aba33d6b0f270f46a6b907b7f5fd5/service/config.go#L62) in the core collector.